fastlane/lanes/release.rb

```diff
@@ -22,25 +22,32 @@
     # Make sure that Gutenberg is configured as expected for a successful code freeze
     gutenberg_dep_check
 
-    # The `release_version_next` is used as the `new internal release version` value because the external and internal
-    # release versions are always the same.
-    message = <<-MESSAGE
-      Code Freeze:
-      • New release branch from #{DEFAULT_BRANCH}: release/#{release_version_next}
+    release_branch_name = compute_release_branch_name(options:, version: release_version_next)
 
-      • Current release version and build code: #{release_version_current} (#{build_code_current}).
-      • New release version and build code: #{release_version_next} (#{build_code_code_freeze}).
+    skip_user_confirmation = options[:skip_confirm]
 
-      • Current internal release version and build code: #{release_version_current_internal} (#{build_code_current_internal})
-      • New internal release version and build code: #{release_version_next} (#{build_code_code_freeze_internal})
-    MESSAGE
+    unless skip_user_confirmation
+      # The `release_version_next` is used as the `new internal release version` value because the external and internal
+      # release versions are always the same.
+      message = <<-MESSAGE
+        Code Freeze:
+        • New release branch from #{DEFAULT_BRANCH}: #{release_branch_name}
 
-    UI.important(message)
-    UI.user_error!('Aborted by user request') unless options[:skip_confirm] || UI.confirm('Do you want to continue?')
+        • Current release version and build code: #{release_version_current} (#{build_code_current}).
+        • New release version and build code: #{release_version_next} (#{build_code_code_freeze}).
+
+        • Current internal release version and build code: #{release_version_current_internal} (#{build_code_current_internal})
+        • New internal release version and build code: #{release_version_next} (#{build_code_code_freeze_internal})
+      MESSAGE
+
+      UI.important(message)
```

Comment:
How do you feel about always printing the `message` outside of the `skip_user_confirmation` check to make the information available in CI logs?

Reply:
My rationale for only printing in interactive mode was that there's no point in showing that message if the user cannot act on it.

But.. your point of making information available in the CI logs is quite valid. Who knows, there might be a bug at some point in how the codes are evaluated, like it happened in https://github.com/wordpress-mobile/WordPress-iOS/pull/22087, at which point having logs might be a huge help in diagnosing.

Addressed in 2f8c867656c240188d0167d1b927b69db281f5d9